### PR TITLE
docs: add MkDocs documentation

### LINKED ISSUE #1105: feature request: add basic usage documentation
### Current Behavior

No idea how to use the tool and what to add to start a session.  Documentation is lacking for someone who has no clue on programming.  Needs to be easy for QA people who may not be super technical

### Suggested Solution

Add documentation to creating a connection to see an emulator or a connected device

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by eglitise**
This PR adds a fully comprehensive documentation for the Inspector. It covers things like quickstart & installation, troubleshooting, contributing guidelines, and of course, descriptions for every bit of functionality in the GUI. Also included are CI workflows for PRs and publishing.
The README will be updated in a separate PR.
Resolves #1105.

### CODE REVIEW

File: docs/quickstart/requirements.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+the Inspector. This server can be run either locally or remotely. If you need to install it, please
+check the [Appium Install documentation](https://appium.io/docs/en/latest/quickstart/install/)
+for details.
+
+!!! note
+
+    If using a standalone Appium server, make sure the server also has the necessary
+    [driver(s)](https://appium.io/docs/en/latest/ecosystem/drivers/) for your target platform(s)!
+
+While the Inspector is designed to work with Appium 2, it is also compatible with
```
Comment (mykola-mokhnach) [inline docs/quickstart/requirements.md:27]: Not sure if we even need to mention Appium 1 except of the fact it has been EOLed a while ago

Comment (eglitise) [reply docs/quickstart/requirements.md:27]: I think it's reasonable to mention here, given that most things in the Appium ecosystem are no longer compatible with Appium 1. There will still be a point where Appium 1-specific workarounds (such as `automationName` not being guaranteed) will be dropped, and the docs updated accordingly.

File: docs/session-builder/capability-builder.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+The right side of the Session Builder also shows the current capabilities, but in the standard JSON
+format, which may be more useful when working with code.
+
+![Capability JSON Structure](assets/images/capability-builder/capability-json.png)
+
+At the top right of the JSON structure, there is an Edit button, which allows adjusting the
+capabilities by directly editing the JSON structure. Toggling this editor mode then shows buttons
+for discarding any changes, or saving them. After saving, any changes will also be reflected in
+[the capability fields](#capability-fields) on the left side of the Session Builder.
+
```
Comment (jlipps) [inline docs/session-builder/capability-builder.md:51]: should we add a note somewhere in this doc that nested objects must be added in raw JSON edit mode? i.e., something like this:

```
{
  "key": {
    "innerKey": "val",
  }
}
```

which is sometimes used for eg `chrome:options` or things like that. we don't have a UI method for building these yet.

**Review by jlipps — APPROVED** [review]
this is a remarkable upgrade to our inspector docs that will greatly assist our users! thank you @eglitise !

File: docs/session-builder/capability-builder.md (hunk 2 above)
Comment (eglitise) [reply docs/session-builder/capability-builder.md:51]: Good idea!

**Pull request merged**